source/val/validate_instruction.cpp

```diff
@@ -379,6 +385,14 @@ spv_result_t LimitCheckStruct(ValidationState_t& _, const Instruction* inst) {
            << "Structure Nesting Depth may not be larger than " << depth_limit
            << ". Found " << cur_depth << ".";
   }
+
+  _.SetHasNestedBlockOrBufferBlockStruct(inst->id(), has_nested_blockOrBufferBlock_struct);
+  if(_.HasNestedBlockOrBufferBlockStruct(inst->id()) && _.HasDecoration(inst->id(), SpvDecorationBlock)){
```

Comment:
This always returns false. Could tell me what am I doing wrong?

Reply:
I think the decorations have not been registered at this point. Your check should be performed later in the flow.

Reply:
That might be it. In that case I think we need to update  _HasDecoration_ definition to return an error, saying it doesn't know yet. Also I couldn't fully understand how this function works. I appreciate it if you could go through it with me when you were in office.